content/blog/2020-12-14-parallel-Java/index.md

```diff
@@ -0,0 +1,126 @@
+
+Thus, the authors choose this method to simplify the parallel programming in Java by introducing deterministic semantics.
+
+## Contribution
+
+(1) The authors describe an region-based type and effect system that provides several new features (RPLs, index-parameterized arrays, subarrays, and invocation effects) over previous systems for expressing deterministic parallel algorithms. These feature guarantee determinism at compile-time.
+
+Specifically, they introduce region path lists (RPLs), which enable more flexible effect summaries including effects on nested structures. They introduce an index-parameterized array type that allows references to provably distinct objects to be stored in an array while still permitting arbitrary aliasing of the objects through references outside the array. They also define the notions of subarrays and partition operations to enable in-place parallel divide and conquer operations on arrays. They also introduce an invocation effect with simple commutativity annotations to permit the parallel invocation of operations that may interfere at the level of reads and write, but still commute logically, i.e., produce the same logical behavior.
+
+(2) The authors develop a formal definition of the static and dynamic semantics and a detailed proof that the system allows sound static inference about noninterference of effects.
+
+(3) The authors describe a language called Deterministic Parallel Java (DPJ) that incorporates the new type and effect system, which supports the full flexibility of the sequential subset of Java.
+
+(4) The authors demonstrate that a practical type of effect system can simplify parallel programming by guaranteeing deterministic semantics with modular, compile-time type checking even in a rich, concurrent object-oriented language such as Java. They study six real-word parallel programs writing in DPJ and shows good performance gains (coming close to or beating equivalent, nondeterministic multithreaded programs where those are available).
+
+## Basic Features in DPJ
+(1) Region names: a filed region declaration declares a new name r (called a field region name), which is associated with the static class.
+
+(2) Region parameters: class and method region parameters which is similar to Java generic parameters. It enables us to create multiple instances of the same class with their data in different regions.
+
+(3) Disjointness constraints: to control aliasing of region parameters. The programmer may write a disjointness constraint of the form P1 # P2, where P1 and P2 are parameters that are required to be disjoint.

```

Comment:
A very short explanation of what you mean by "disjoint" might be useful. Perhaps via an example? (Just a sentence is fine.)